src/main/java/com/openlattice/chronicle/services/download/ParticipantDataIterable.kt

```diff
@@ -119,52 +160,33 @@ class ParticipantDataIterable(private val supplier: NeighborPageSupplier) : Iter
-                            .iterator()
-                            .next()
-                            .toString()
+                    val neighborValues = getDownloadValuesForEntity(
+                            neighborEntityDetails[currentIndex].neighborDetails.get(),
+                            neighborPageSupplier.srcMetadata,
+                            neighborPageSupplier.srcPropertiesToInclude,
+                            OutputConstants.APP_PREFIX
+                    )
+
+                    val associationValues = getDownloadValuesForEntity(
+                            neighborEntityDetails[currentIndex].associationDetails,
+                            neighborPageSupplier.edgeMetadata,
+                            neighborPageSupplier.edgePropertiesToInclude,
+                            OutputConstants.USER_PREFIX
                     )
-                    entities.forEach { (key: FullQualifiedName?, `val`: Set<Any?>) ->
-                        if (!neighborPageSupplier.srcPropertiesToExclude.contains(key)) {
-                            val propertyType = neighborPageSupplier.edmCacheManager
-                                    .getPropertyType(key)
-                            val title = OutputConstants.APP_PREFIX + neighborPageSupplier.srcMetadata.getValue(propertyType.id)
-                                    .title
-                            if (propertyType.datatype == EdmPrimitiveTypeKind.DateTimeOffset) {
-                                nextElement[title] = parseDateTimeValues(`val`, zoneId)
-                            } else {
-                                nextElement[title] = `val`
-                            }
-                        }
-                    }
 
-                    // 2: process association data
-                    val associations = neighborEntityDetails[currentIndex]
-                            .associationDetails
-                    associations.forEach { (key: FullQualifiedName?, `val`: Set<Any?>) ->
-                        if (!neighborPageSupplier.edgePropertiesToExclude.contains(key)) {
-                            val propertyTypeId = neighborPageSupplier.edmCacheManager
-                                    .getPropertyTypeId(key)
-                            val title = OutputConstants.USER_PREFIX + neighborPageSupplier.edgeMetadata.getValue(propertyTypeId)
-                                    .title
-                            nextElement[title] = `val`
-                        }
-                    }
                     currentIndex++
                     finishedCurrentPage = currentIndex == neighborEntityDetails.size
+
+                    return neighborValues + associationValues
                 }
             } catch (e: Exception) {
                 logger.error("unable to retrieve next element", e)
+                return mapOf()
             }
-
-            return nextElement.toMap()
+            return mapOf()
```

Comment:
Should there be some logging here? I can't quite tell if this code will ever get reached 🤔

Reply:
yeah in this case this will never be executed.

I will follow up in a separate pr just in case